beefy-gadget/src/keystore.rs

```diff
@@ -24,6 +24,19 @@ use beefy_primitives::{
 
 use crate::error;
 
+// Keystore newtype helper
+//
+// Return wrapped keystore instance or an error in case there is no keystore.
+macro_rules! keystore {
```

Comment:
Why a macro and not a function?

Reply:
A function wouldn't add any benefit, since the error handling would still be explicit. The idea was to hide the somewhat verbose error handling. That said, the idiomatic version outlined by you should be good enough here.